types/index.d.ts

```diff
@@ -7,14 +7,16 @@ import { SvelteComponentTyped } from 'svelte/types/runtime'
 
 export * from '@testing-library/dom'
 
-export interface SvelteComponentOptions<P extends Record<string, any> = any>  {
+type SvelteComponentTypedClass<P,E,S> = typeof SvelteComponentTyped<P,E,S>;
```

Comment:
Yeah so for some reason, `typeof` doesn't really play nicely when it comes to generics. I'm not sure if TypeScript gives an escape hatch for that. But what's currently here may not work.

Reply:
This doesn't work for me either, TS cannot parse this. If I understand correctly, the class definition should be something like

```typescript
type SvelteComponentTypedClass<P, E, S> = new (options: SvelteComponentOptions<P>) => SvelteComponentTyped<P, E, S>;
```

See also svelte2tsx:
https://github.com/sveltejs/language-tools/blob/1415196df7118f4361385a6450ff2756bca3ff00/packages/svelte2tsx/svelte-shims.d.ts#L56-L76